runtime/near-vm-errors/src/lib.rs

```diff
@@ -57,7 +56,6 @@ pub enum FunctionCallErrorSer {
     WasmTrap(WasmTrap),
     WasmUnknownError,
     HostError(HostError),
-    EvmError(EvmError),
```

Comment:
Wait, as this wasnt' gated behind a feature, this now changes serialization format, which is a breaking change. Let's do the standard trick and rename this variant to `Unused` to preserve borsh serialization format. Let's also add a comment that this variant was, in fact, *never* used, so that we could re-use this variant in the future. 

To bad that we didn't make this variant last to begin with. Had we done so, just dropping it would have been correct.

Reply:
Could you double check that this is the single place where we need to think about compatibility?

Reply:
how come this was not gated by a feature :(

Reply:
@matklad I double checked there is another enum that not place evmerror as last, which need to be replaced by `_EVMError` to keep borsh correct. Also double checked every file else don't see any compatibility issue.

@bowenwang1996 Evm error was added first then evm code & feature. I think we were sure that evm will be landed so just guard evm logic code was sufficient. We didn't consider the case that evm would be removed